docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+
+```bash
+MySubtopology/
+├─ CMakeLists.txt
+├─ MySubtopology.cmake
+├─ MySubtopology.fpp
+├─ MySubtopologyTopology.cpp
+├─ MySubtopologyTopology.hpp
+└─ MySubtopologyTopologyDefs.hpp
+```
+
+All files will be discussed in more detail in later sections of this guide. Additionally, note that there are optional files that can be included in your subtopology to extend its capability.
+
+- It is highly recommended to include a `docs` folder to document your subtopology. Simple markdown files work well in this case.
+- `.fppi` files can be included as config files to your subtopology. They can include useful constants or structures that allow users to modify your subtopology.
+- Unit tests can also be added to your subtopology, using a similar structure to unit tests for components.
+- Note that it is possible to write your subtopology with [phases](https://nasa.github.io/fpp/fpp-users-guide.html#Defining-Component-Instances_Init-Specifiers_Execution-Phases). The example repository in the [last section](#conclusion) includes an example of this in the "example/phases" branch.
+
+> Note that with the latest release of `fprime-tools`, you can run `fprime-util new --subtopology` to generate the subtopology structure.
+
+## Individual File Contents
+
+This section will provide an overview to the contents of the individual files that make up a subtopology. We will use the names of the required files structure as shown in the [above](#subtopology-structure) section. However, note that the file names shown are not required to be duplicated within your own custom subtopologies.
+
+### Example Scenario
+
+For this section, it would be helpful to come up with an example scenario where the subtopology is used, such that it can be better understood. Let's imagine that we would like to create a component called `RNG` that will write a random integer to some telemetry channel when it is hooked up to a 1Hz clock (rate group). The `RNG` component will have an input port called `run`, that will be the entry point from the clock (rate group). The `RNG` component will be an active component, and is under the `MyLibrary` namespace.
+
+### MySubtopology.fpp
+
+As mentioned, this is a unified `.fpp` file. It not only includes the instances instance declarations for components that you'd like your subtopology to use, but also the definitions for the instances.
+
+Based on our example, we may have the following instance declarations:
+
+```cpp
+instance rng // RNG component instance
+instance rateGroup // rate group instance
+``` 
+
+and thus the following instance definitions:
+
+```cpp
+instance rng: MyLibrary.RNG base id 0xFF2FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size Defaults.STACK_SIZE \
+    priority 100
+
+instance rateGroup: Svc.ActiveRateGroup base id 0xFF4FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size DEFAULTS.STACK_SIZE \
+    priority 150
+
+instance rateGroupDriver: Svc.RateGroupDriver base id 0xFF8FF // to drive the 1Hz rate group
```

Comment:
Since `rateGroupDriver` is not going to be in the subtopology but rather hooked from the Main topology, would it be good to remove it from the instance list here? And rather add a note right below to emphasize that this is a feature of subtopologies, leverage components from outside that the user will wire up to their subtopology.